diff --git a/lib/rules_block/blockquote.mjs b/lib/rules_block/blockquote.mjs
index b61da02ec9603b04929246f6df0490bcc0a07c7b..c1bc59565aba9d836b10c6b0f2b5f7184742001f 100644
--- a/lib/rules_block/blockquote.mjs
+++ b/lib/rules_block/blockquote.mjs
@@ -9,7 +9,7 @@ export default function blockquote (state, startLine, endLine, silent) {
   const oldLineMax = state.lineMax
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   // check the block quote marker
   if (state.src.charCodeAt(pos) !== 0x3E/* > */) { return false }
diff --git a/lib/rules_block/code.mjs b/lib/rules_block/code.mjs
index e45e6f9ace655b3c20a349c9d8c779f4979e3602..fffd7378c6aa26de78d2134de78561e2c64a2bdc 100644
--- a/lib/rules_block/code.mjs
+++ b/lib/rules_block/code.mjs
@@ -1,6 +1,7 @@
 // Code block (4 spaces padded)
 
 export default function code (state, startLine, endLine/*, silent */) {
+  if (state.md.options.allowIndentation) { return false }
   if (state.sCount[startLine] - state.blkIndent < 4) { return false }
 
   let nextLine = startLine + 1
diff --git a/lib/rules_block/fence.mjs b/lib/rules_block/fence.mjs
index 930f7b364e325b31c96685f04f91d956a93efb83..0e8128225be44e3f43d511142fc38095040b5860 100644
--- a/lib/rules_block/fence.mjs
+++ b/lib/rules_block/fence.mjs
@@ -5,7 +5,7 @@ export default function fence (state, startLine, endLine, silent) {
   let max = state.eMarks[startLine]
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   if (pos + 3 > max) { return false }
 
@@ -59,7 +59,7 @@ export default function fence (state, startLine, endLine, silent) {
 
     if (state.src.charCodeAt(pos) !== marker) { continue }
 
-    if (state.sCount[nextLine] - state.blkIndent >= 4) {
+    if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent >= 4) {
       // closing fence should be indented less than 4 spaces
       continue
     }
diff --git a/lib/rules_block/heading.mjs b/lib/rules_block/heading.mjs
index d2f7b79f66fe38a67d1956e546a72745c94d9c48..c5b08c0064cfcb5ff5ffbdc1a796367e241b8ac6 100644
--- a/lib/rules_block/heading.mjs
+++ b/lib/rules_block/heading.mjs
@@ -7,7 +7,7 @@ export default function heading (state, startLine, endLine, silent) {
   let max = state.eMarks[startLine]
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   let ch  = state.src.charCodeAt(pos)
 
diff --git a/lib/rules_block/hr.mjs b/lib/rules_block/hr.mjs
index d467b21497a07951a742889ea2ed74a26a38c3f8..d3e6e07e6d30650c449ef62c6fc3688c025918d9 100644
--- a/lib/rules_block/hr.mjs
+++ b/lib/rules_block/hr.mjs
@@ -5,7 +5,7 @@ import { isSpace } from '../common/utils.mjs'
 export default function hr (state, startLine, endLine, silent) {
   const max = state.eMarks[startLine]
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   let pos = state.bMarks[startLine] + state.tShift[startLine]
   const marker = state.src.charCodeAt(pos++)
diff --git a/lib/rules_block/html_block.mjs b/lib/rules_block/html_block.mjs
index 197520f67b15413733fc6b0fef7248fee1108086..e839577cde2b5319ba43cd266d1b1c1536444297 100644
--- a/lib/rules_block/html_block.mjs
+++ b/lib/rules_block/html_block.mjs
@@ -21,7 +21,7 @@ export default function html_block (state, startLine, endLine, silent) {
   let max = state.eMarks[startLine]
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   if (!state.md.options.html) { return false }
 
diff --git a/lib/rules_block/lheading.mjs b/lib/rules_block/lheading.mjs
index ee3b9a303ea8bec384eef93d8c469f4658933e6a..d9b9b65a276e1d1fa39fbf6f2ea480feaa2bea0e 100644
--- a/lib/rules_block/lheading.mjs
+++ b/lib/rules_block/lheading.mjs
@@ -4,7 +4,7 @@ export default function lheading (state, startLine, endLine/*, silent */) {
   const terminatorRules = state.md.block.ruler.getRules('paragraph')
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   const oldParentType = state.parentType
   state.parentType = 'paragraph' // use paragraph to match terminatorRules
@@ -17,7 +17,7 @@ export default function lheading (state, startLine, endLine/*, silent */) {
   for (; nextLine < endLine && !state.isEmpty(nextLine); nextLine++) {
     // this would be a code block normally, but after paragraph
     // it's considered a lazy continuation regardless of what's there
-    if (state.sCount[nextLine] - state.blkIndent > 3) { continue }
+    if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent > 3) { continue }
 
     //
     // Check for underline in setext header
diff --git a/lib/rules_block/list.mjs b/lib/rules_block/list.mjs
index fb53abd49ef04a2f7688e5435f523235f2a11fa2..28e99f7f241113d906890948290efb5854ba8e1f 100644
--- a/lib/rules_block/list.mjs
+++ b/lib/rules_block/list.mjs
@@ -93,7 +93,7 @@ export default function list (state, startLine, endLine, silent) {
   let tight = true
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[nextLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent >= 4) { return false }
 
   // Special case:
   //  - item 1
@@ -101,7 +101,8 @@ export default function list (state, startLine, endLine, silent) {
   //    - item 3
   //     - item 4
   //      - this one is a paragraph continuation
-  if (state.listIndent >= 0 &&
+  if (!state.md.options.allowIndentation &&
+      state.listIndent >= 0 &&
       state.sCount[nextLine] - state.listIndent >= 4 &&
       state.sCount[nextLine] < state.blkIndent) {
     return false
@@ -211,7 +212,7 @@ export default function list (state, startLine, endLine, silent) {
 
     // If we have more than 4 spaces, the indent is 1
     // (the rest is just indented code block)
-    if (indentAfterMarker > 4) { indentAfterMarker = 1 }
+    if (!state.md.options.allowIndentation && indentAfterMarker > 4) { indentAfterMarker = 1 }
 
     // "  -  test"
     //  ^^^^^ - calculating total length of this thing
@@ -284,7 +285,7 @@ export default function list (state, startLine, endLine, silent) {
     if (state.sCount[nextLine] < state.blkIndent) { break }
 
     // if it's indented more than 3 spaces, it should be a code block
-    if (state.sCount[nextLine] - state.blkIndent >= 4) { break }
+    if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent >= 4) { break }
 
     // fail if terminating block found
     let terminate = false
diff --git a/lib/rules_block/paragraph.mjs b/lib/rules_block/paragraph.mjs
index 6ecdcef0d3f1dc95b945de44ed2f7a8512e11daa..4266ab37238b11559d2e3128e92e60fe1857d577 100644
--- a/lib/rules_block/paragraph.mjs
+++ b/lib/rules_block/paragraph.mjs
@@ -10,7 +10,7 @@ export default function paragraph (state, startLine, endLine) {
   for (; nextLine < endLine && !state.isEmpty(nextLine); nextLine++) {
     // this would be a code block normally, but after paragraph
     // it's considered a lazy continuation regardless of what's there
-    if (state.sCount[nextLine] - state.blkIndent > 3) { continue }
+    if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent > 3) { continue }
 
     // quirk for blockquotes, this line should already be checked by that rule
     if (state.sCount[nextLine] < 0) { continue }
diff --git a/lib/rules_block/reference.mjs b/lib/rules_block/reference.mjs
index 416628645a5fa0796aad19d902d7d1d8307b9a84..41e7e6e1a95a9b875cc3faf3a9cca057817128f6 100644
--- a/lib/rules_block/reference.mjs
+++ b/lib/rules_block/reference.mjs
@@ -6,7 +6,7 @@ export default function reference (state, startLine, _endLine, silent) {
   let nextLine = startLine + 1
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
 
   if (state.src.charCodeAt(pos) !== 0x5B/* [ */) { return false }
 
@@ -22,7 +22,7 @@ export default function reference (state, startLine, _endLine, silent) {
 
     // this would be a code block normally, but after paragraph
     // it's considered a lazy continuation regardless of what's there
-    if (state.sCount[nextLine] - state.blkIndent > 3) { isContinuation = true }
+    if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent > 3) { isContinuation = true }
 
     // quirk for blockquotes, this line should already be checked by that rule
     if (state.sCount[nextLine] < 0) { isContinuation = true }
diff --git a/lib/rules_block/table.mjs b/lib/rules_block/table.mjs
index be0ba0a26a432c156fc02ce2c71fdc5743ca0a5d..f0760a93c08e087098ab2addc56ba86bb39f403e 100644
--- a/lib/rules_block/table.mjs
+++ b/lib/rules_block/table.mjs
@@ -61,7 +61,7 @@ export default function table (state, startLine, endLine, silent) {
   if (state.sCount[nextLine] < state.blkIndent) { return false }
 
   // if it's indented more than 3 spaces, it should be a code block
-  if (state.sCount[nextLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent >= 4) { return false }
 
   // first character of the second line should be '|', '-', ':',
   // and no other characters are allowed but spaces;
@@ -119,7 +119,7 @@ export default function table (state, startLine, endLine, silent) {
 
   lineText = getLine(state, startLine).trim()
   if (lineText.indexOf('|') === -1) { return false }
-  if (state.sCount[startLine] - state.blkIndent >= 4) { return false }
+  if (!state.md.options.allowIndentation && state.sCount[startLine] - state.blkIndent >= 4) { return false }
   columns = escapedSplit(lineText)
   if (columns.length && columns[0] === '') columns.shift()
   if (columns.length && columns[columns.length - 1] === '') columns.pop()
@@ -181,7 +181,7 @@ export default function table (state, startLine, endLine, silent) {
     if (terminate) { break }
     lineText = getLine(state, nextLine).trim()
     if (!lineText) { break }
-    if (state.sCount[nextLine] - state.blkIndent >= 4) { break }
+    if (!state.md.options.allowIndentation && state.sCount[nextLine] - state.blkIndent >= 4) { break }
     columns = escapedSplit(lineText)
     if (columns.length && columns[0] === '') columns.shift()
     if (columns.length && columns[columns.length - 1] === '') columns.pop()
